build is not working anymore due to missing dependency

```
$ ./gradlew shadowDistZip

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':shadowJar'.
> Could not resolve all files for configuration ':runtimeClasspath'.
   > Could not download jai_core-1.1.3.jar (javax.media:jai_core:1.1.3)
      > Could not get resource 'https://maven.geotoolkit.org/javax/media/jai_core/1.1.3/jai_core-1.1.3.jar'.
         > Could not GET 'https://maven.geotoolkit.org/javax/media/jai_core/1.1.3/jai_core-1.1.3.jar'.
            > The server may not support the client's requested TLS protocol versions: (TLSv1.2, TLSv1.3). You may need to configure the client to allow other protocols to be used. See: https://docs.gradle.org/7.6/userguide/build_environment.html#gradle_system_properties
               > PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
   > Could not download jai_codec-1.1.3.jar (javax.media:jai_codec:1.1.3)
      > Could not get resource 'https://maven.geotoolkit.org/javax/media/jai_codec/1.1.3/jai_codec-1.1.3.jar'.
         > Could not GET 'https://maven.geotoolkit.org/javax/media/jai_codec/1.1.3/jai_codec-1.1.3.jar'.
            > The server may not support the client's requested TLS protocol versions: (TLSv1.2, TLSv1.3). You may need to configure the client to allow other protocols to be used. See: https://docs.gradle.org/7.6/userguide/build_environment.html#gradle_system_properties
               > PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
```


Adding https://repository.jboss.org/nexus/content/repositories/thirdparty-releases to the repositories doesn't help either:

```
$ ./gradlew shadowDistZip
Starting a Gradle Daemon, 3 incompatible Daemons could not be reused, use --status for details

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':shadowJar'.
> Could not resolve all dependencies for configuration ':runtimeClasspath'.
   > Could not find javax.media:jai_codec:1.1.3.
     Searched in the following locations:
       - https://clojars.org/repo/javax/media/jai_codec/1.1.3/jai_codec-1.1.3.pom
       - https://repository.jboss.org/nexus/content/repositories/thirdparty-releases/javax/media/jai_codec/1.1.3/jai_codec-1.1.3.pom
       - https://repo.maven.apache.org/maven2/javax/media/jai_codec/1.1.3/jai_codec-1.1.3.pom
     Required by:
         project : > org.jpedal:jpedal-lgpl:4.74b27

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s
```

Seemingly, the jpedal's dependencies have disappeared from public repositories.

--- a/.github/workflows/gradle.yml
+++ b/.github/workflows/gradle.yml
@@ -5,9 +5,9 @@ name: Java CI with Gradle
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
   pull_request:
-    branches: [ master ]
+    branches: [master]
   workflow_dispatch:
 
 jobs:
@@ -15,29 +15,29 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java-version: [ 1.8, 1.11, 1.17, 1.19 ]
+        java-version:
+          - 8.0.462+8
+          - 11.0.28+6
+          - 17.0.16+8
+          - 19.0.2+7
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK ${{ matrix.java-version }}
-      uses: actions/setup-java@v1
-      with:
-        java-version: ${{ matrix.java-version }}
-        distribution: 'temurin'
-    - name: Cache Gradle packages
-      uses: actions/cache@v1
-      with:
-        path: ~/.gradle/caches
-        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
-        restore-keys: ${{ runner.os }}-gradle
-    - name: Verify formatting
-      run: ./gradlew spotlessCheck
-    - name: Create bundle
-      run: ./gradlew shadowDistZip
-    - name: Run automated tests
-      run: ./gradlew test
-    - name: Archive package
-      uses: actions/upload-artifact@v3
-      with:
-        name: Briss-2.0-${{ matrix.java-version }}
-        path: build/distributions/*.zip
+      - uses: actions/checkout@v4
+      - name: Set up JDK ${{ matrix.java-version }}
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ matrix.java-version }}
+          distribution: "temurin"
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+      - name: Verify formatting
+        run: ./gradlew spotlessCheck
+      - name: Create bundle
+        run: ./gradlew shadowDistZip
+      - name: Run automated tests
+        run: ./gradlew test
+      - name: Archive package
+        uses: actions/upload-artifact@v4
+        with:
+          name: Briss-2.0-${{ matrix.java-version }}
+          path: build/distributions/*.zip

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased]
+* [Issue-91] Migrate from JPedal to Apache PDFBox since the former has been removed from public repositories.
+
 ## [2.0-alpha 5]
 
 ### Added

--- a/build.gradle
+++ b/build.gradle
@@ -10,16 +10,15 @@ mainClassName = 'at.laborg.briss.Briss'
 applicationDefaultJvmArgs = ["-Xms128m", "-Xmx1024m"]
 
 repositories {
-    maven { url "https://clojars.org/repo" } // For jpedal
-    maven { url "https://maven.geotoolkit.org/" } // For jpedal transitive dependencies
     mavenCentral()
 }
 
 dependencies {
     implementation 'com.itextpdf:itextpdf:5.5.13.4'
-    implementation 'org.jpedal:jpedal-lgpl:4.74b27'
     implementation 'org.bouncycastle:bcprov-jdk18on:1.79'
     implementation 'org.bouncycastle:bcpkix-jdk18on:1.79'
+    implementation 'org.apache.pdfbox:pdfbox:3.0.5'
+    implementation 'org.apache.pdfbox:pdfbox-tools:3.0.5'
 
 
     testImplementation(platform('org.junit:junit-bom:5.10.3'))

--- a/src/main/java/at/laborg/briss/BrissGUIApp.java
+++ b/src/main/java/at/laborg/briss/BrissGUIApp.java
@@ -2,7 +2,6 @@ package at.laborg.briss;
 
 import java.io.File;
 import java.io.IOException;
-import org.jpedal.exception.PdfException;
 
 public interface BrissGUIApp {
 
@@ -16,5 +15,5 @@ public interface BrissGUIApp {
 
 	void resizeSelRects(int w, int h);
 
-	void importNewPdfFile(File loadFile) throws IOException, PdfException;
+	void importNewPdfFile(File loadFile) throws IOException;
 }

--- a/src/main/java/at/laborg/briss/BrissSwingGUI.java
+++ b/src/main/java/at/laborg/briss/BrissSwingGUI.java
@@ -82,7 +82,6 @@ import javax.swing.UIManager;
 import javax.swing.UnsupportedLookAndFeelException;
 import javax.swing.event.AncestorEvent;
 import javax.swing.event.AncestorListener;
-import org.jpedal.exception.PdfException;
 
 /**
  * @author gerhard, hybridtupel
@@ -155,7 +154,7 @@ public class BrissSwingGUI implements BrissGUIApp {
 		if (fileArg.exists() && fileArg.getAbsolutePath().trim().endsWith(".pdf")) { // $NON-NLS-1$
 			try {
 				importNewPdfFile(fileArg);
-			} catch (IOException | PdfException e) {
+			} catch (IOException e) {
 				JOptionPane.showMessageDialog(mainWindow, e.getMessage(), Messages.getString("BrissGUI.brissError"),
 						JOptionPane.ERROR_MESSAGE); // $NON-NLS-1$
 			}
@@ -351,7 +350,7 @@ public class BrissSwingGUI implements BrissGUIApp {
 		} catch (IOException e) {
 			JOptionPane.showMessageDialog(mainWindow, e.getMessage(), Messages.getString("BrissGUI."),
 					JOptionPane.ERROR_MESSAGE); // $NON-NLS-1$
-		} catch (PdfException e) {
+		} catch (Exception e) {
 			JOptionPane.showMessageDialog(mainWindow, e.getMessage(), Messages.getString("BrissGUI.loadingError"),
 					JOptionPane.ERROR_MESSAGE); // $NON-NLS-1$
 		}
@@ -409,7 +408,7 @@ public class BrissSwingGUI implements BrissGUIApp {
 		mainWindow.setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
 	}
 
-	public void importNewPdfFile(File loadFile) throws IOException, PdfException {
+	public void importNewPdfFile(File loadFile) throws IOException {
 		String password = null;
 
 		if (PDFReaderUtil.isEncrypted(loadFile.getAbsolutePath())) {

--- a/src/main/java/at/laborg/briss/BrissTransferHandler.java
+++ b/src/main/java/at/laborg/briss/BrissTransferHandler.java
@@ -25,7 +25,6 @@ import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
 import javax.swing.TransferHandler;
-import org.jpedal.exception.PdfException;
 
 final class BrissTransferHandler extends TransferHandler {
 
@@ -69,7 +68,7 @@ final class BrissTransferHandler extends TransferHandler {
 					}
 					try {
 						this.brissGUI.importNewPdfFile(loadFile);
-					} catch (PdfException e) {
+					} catch (Exception e) {
 						// TODO Auto-generated catch block
 						e.printStackTrace();
 					}

--- a/src/main/java/at/laborg/briss/ClusterManager.java
+++ b/src/main/java/at/laborg/briss/ClusterManager.java
@@ -85,17 +85,17 @@ public class ClusterManager {
 		public void run() {
 			File source = new File(clusterJob.getSource().getAbsolutePath());
 
-            try (PDFImageExtractor pdfImageExtractor = new PDFImageExtractor(source, null)) {
-                for (SingleCluster cluster : clusterJob.getClusterCollection().getAsList()) {
-                    for (Integer pageNumber : cluster.getPagesToMerge()) {
-                        BufferedImage renderedPage = pdfImageExtractor.extractImage(pageNumber - 1);
-                        cluster.getImageData().addImageToPreview(renderedPage);
-                        workerUnitCounter++;
-                    }
-                }
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+			try (PDFImageExtractor pdfImageExtractor = new PDFImageExtractor(source, null)) {
+				for (SingleCluster cluster : clusterJob.getClusterCollection().getAsList()) {
+					for (Integer pageNumber : cluster.getPagesToMerge()) {
+						BufferedImage renderedPage = pdfImageExtractor.extractImage(pageNumber - 1);
+						cluster.getImageData().addImageToPreview(renderedPage);
+						workerUnitCounter++;
+					}
+				}
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
 		}
 	}
 }

--- a/src/main/java/at/laborg/briss/ClusterManager.java
+++ b/src/main/java/at/laborg/briss/ClusterManager.java
@@ -20,17 +20,17 @@ package at.laborg.briss;
 import at.laborg.briss.model.ClusterCollection;
 import at.laborg.briss.model.ClusterJob;
 import at.laborg.briss.model.SingleCluster;
+import at.laborg.briss.utils.PDFImageExtractor;
 import com.itextpdf.text.Rectangle;
 import com.itextpdf.text.pdf.PdfReader;
+
 import java.awt.image.BufferedImage;
 import java.io.File;
 import java.io.IOException;
-import org.jpedal.PdfDecoder;
-import org.jpedal.exception.PdfException;
 
 public class ClusterManager {
 
-	public static ClusterJob createClusterJob(File origFile) throws IOException, PdfException {
+	public static ClusterJob createClusterJob(File origFile) throws IOException {
 
 		PdfReader reader = new PdfReader(origFile.getAbsolutePath());
 		ClusterJob clusterJob = new ClusterJob(origFile);
@@ -83,27 +83,19 @@ public class ClusterManager {
 
 		@Override
 		public void run() {
-			PdfDecoder pdfDecoder = new PdfDecoder(true);
-			try {
-				pdfDecoder.openPdfFile(clusterJob.getSource().getAbsolutePath());
-			} catch (PdfException e1) {
-				e1.printStackTrace();
-			}
+			File source = new File(clusterJob.getSource().getAbsolutePath());
 
-			for (SingleCluster cluster : clusterJob.getClusterCollection().getAsList()) {
-				for (Integer pageNumber : cluster.getPagesToMerge()) {
-					try {
-						BufferedImage renderedPage = pdfDecoder.getPageAsImage(pageNumber);
-						cluster.getImageData().addImageToPreview(renderedPage);
-						workerUnitCounter++;
-					} catch (PdfException e) {
-						// TODO Auto-generated catch block
-						e.printStackTrace();
-					}
-				}
-			}
-			// now close the reader as it's not used anymore
-			pdfDecoder.closePdfFile();
+            try (PDFImageExtractor pdfImageExtractor = new PDFImageExtractor(source, null)) {
+                for (SingleCluster cluster : clusterJob.getClusterCollection().getAsList()) {
+                    for (Integer pageNumber : cluster.getPagesToMerge()) {
+                        BufferedImage renderedPage = pdfImageExtractor.extractImage(pageNumber - 1);
+                        cluster.getImageData().addImageToPreview(renderedPage);
+                        workerUnitCounter++;
+                    }
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
 		}
 	}
 }

--- a/src/main/java/at/laborg/briss/utils/ClusterRenderWorker.java
+++ b/src/main/java/at/laborg/briss/utils/ClusterRenderWorker.java
@@ -7,8 +7,6 @@ import at.laborg.briss.model.ClusterDefinition;
 import at.laborg.briss.model.PageCluster;
 import java.awt.image.BufferedImage;
 import java.io.File;
-import org.jpedal.PdfDecoder;
-import org.jpedal.exception.PdfException;
 
 public class ClusterRenderWorker extends Thread {
 
@@ -26,27 +24,16 @@ public class ClusterRenderWorker extends Thread {
 
 	@Override
 	public final void run() {
-		PdfDecoder pdfDecoder = new PdfDecoder();
-
-		try {
-			pdfDecoder.openPdfFile(source.getAbsolutePath(), password);
-		} catch (PdfException e1) {
-			e1.printStackTrace();
-		}
-
-		for (PageCluster cluster : clusters.getClusterList()) {
-			for (Integer pageNumber : cluster.getPagesToMerge()) {
-				try {
-					BufferedImage renderedPage = pdfDecoder.getPageAsImage(pageNumber);
-					cluster.getImageData().addImageToPreview(renderedPage);
-					workerUnitCounter++;
-				} catch (PdfException e) {
-					// TODO Auto-generated catch block
-					e.printStackTrace();
-				}
-			}
-		}
-		// now close the reader as it's not used anymore
-		pdfDecoder.closePdfFile();
+		try (PDFImageExtractor pdfImageExtractor = new PDFImageExtractor(source, password)) {
+            for (PageCluster cluster : clusters.getClusterList()) {
+                for (Integer pageNumber : cluster.getPagesToMerge()) {
+                    BufferedImage renderedPage = pdfImageExtractor.extractImage(pageNumber - 1);
+                    cluster.getImageData().addImageToPreview(renderedPage);
+                    workerUnitCounter++;
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
 	}
 }

--- a/src/main/java/at/laborg/briss/utils/ClusterRenderWorker.java
+++ b/src/main/java/at/laborg/briss/utils/ClusterRenderWorker.java
@@ -25,15 +25,15 @@ public class ClusterRenderWorker extends Thread {
 	@Override
 	public final void run() {
 		try (PDFImageExtractor pdfImageExtractor = new PDFImageExtractor(source, password)) {
-            for (PageCluster cluster : clusters.getClusterList()) {
-                for (Integer pageNumber : cluster.getPagesToMerge()) {
-                    BufferedImage renderedPage = pdfImageExtractor.extractImage(pageNumber - 1);
-                    cluster.getImageData().addImageToPreview(renderedPage);
-                    workerUnitCounter++;
-                }
-            }
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+			for (PageCluster cluster : clusters.getClusterList()) {
+				for (Integer pageNumber : cluster.getPagesToMerge()) {
+					BufferedImage renderedPage = pdfImageExtractor.extractImage(pageNumber - 1);
+					cluster.getImageData().addImageToPreview(renderedPage);
+					workerUnitCounter++;
+				}
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
 	}
 }

--- a/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
+++ b/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
@@ -9,7 +9,7 @@ import java.awt.image.BufferedImage;
 import java.io.File;
 import java.io.IOException;
 
-public class PDFImageExtractor implements AutoCloseable{
+public class PDFImageExtractor implements AutoCloseable {
     private final PDDocument document;
     private final PDFRenderer pdfRenderer;
 

--- a/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
+++ b/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
@@ -1,0 +1,31 @@
+package at.laborg.briss.utils;
+
+import org.apache.pdfbox.Loader;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.rendering.ImageType;
+import org.apache.pdfbox.rendering.PDFRenderer;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+public class PDFImageExtractor implements AutoCloseable{
+    private final PDDocument document;
+    private final PDFRenderer pdfRenderer;
+
+    public PDFImageExtractor(File pdfFile, String password) throws IOException {
+        this.document = Loader.loadPDF(pdfFile, password);
+
+        this.pdfRenderer = new PDFRenderer(document);
+    }
+
+    public BufferedImage extractImage(int pageNumber) throws IOException {
+        return pdfRenderer.renderImageWithDPI(
+            pageNumber, 300, ImageType.RGB);
+    }
+
+    @Override
+    public void close() throws Exception {
+        this.document.close();
+    }
+}

--- a/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
+++ b/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
@@ -10,22 +10,21 @@ import java.io.File;
 import java.io.IOException;
 
 public class PDFImageExtractor implements AutoCloseable {
-    private final PDDocument document;
-    private final PDFRenderer pdfRenderer;
+	private final PDDocument document;
+	private final PDFRenderer pdfRenderer;
 
-    public PDFImageExtractor(File pdfFile, String password) throws IOException {
-        this.document = Loader.loadPDF(pdfFile, password);
+	public PDFImageExtractor(File pdfFile, String password) throws IOException {
+		this.document = Loader.loadPDF(pdfFile, password);
 
-        this.pdfRenderer = new PDFRenderer(document);
-    }
+		this.pdfRenderer = new PDFRenderer(document);
+	}
 
-    public BufferedImage extractImage(int pageNumber) throws IOException {
-        return pdfRenderer.renderImageWithDPI(
-            pageNumber, 300, ImageType.RGB);
-    }
+	public BufferedImage extractImage(int pageNumber) throws IOException {
+		return pdfRenderer.renderImageWithDPI(pageNumber, 300, ImageType.RGB);
+	}
 
-    @Override
-    public void close() throws IOException {
-        this.document.close();
-    }
+	@Override
+	public void close() throws IOException {
+		this.document.close();
+	}
 }

--- a/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
+++ b/src/main/java/at/laborg/briss/utils/PDFImageExtractor.java
@@ -25,7 +25,7 @@ public class PDFImageExtractor implements AutoCloseable {
     }
 
     @Override
-    public void close() throws Exception {
+    public void close() throws IOException {
         this.document.close();
     }
 }

